src/app/pages/wallets/my-wallets/pages-wallets-my-wallets.service.ts

```diff
@@ -16,4 +16,16 @@ export class PagesWalletsMyWalletsService {
       map(walletsResp => walletsResp.map(resp => new MyWallet(resp)))
     );
   }
+
+  public addNewWallet(name_: string): Observable<MyWallet> {
```

Comment:
You can basically have only one method called `openWalletModal(wallet?: MyWallet)`. Notice that `wallet` parameter is optional.In case of adding new wallet you will call this method without argument.

And now in this method you can:
- first open modal with or without wallet passed to this modal (depends if you are adding or editing wallet)
- modal will return (or not if it was canceled) updated / created data of the wallet
- you can use this data to create / update wallet using gateway
- you can return created / updated wallet from this method to the component calling this method (or nothing if modal was canceled)